src/util/device.ts

```diff
@@ -11,3 +12,33 @@ export const isProvisioned = (
 		device.last_connectivity_event != null
 	);
 };
+
+/**
+ *
+ * @param targetState
+ * @returns array containing all images for all services for all releases for all apps for the device
+ */
+export const listImagesFromTargetState = (
```

Comment:
If this is not used by the SDK at all, why do we have it here? Which are the clients will be consuming this?

Reply:
`balena-img` will consume this

Reply:
If it is only `balena-img` will be consuming this, then it doesn't need to be part of the sdk's exposed API, and moving it closer to/in the consumer imo makes more sense.

But TBH even if the CLI would also need to use this, imo it's just 4 nested for loops which is trivial to inline.
Btw
See: https://github.com/balena-io/balena-sdk/pull/1347

Reply:
In other words, I would imagine adding a comment like `// don't remove this b/c the img-maker needs it` would not look great. For example maybe preload could have such a helper or even better use it internally for its own needs and expose a higher level method.